incident/incident.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package incident provides access to functionality for recording and
+// classifying compliance incidents.
+package incident
+
+import (
+	"context"
+
+	"github.com/golang/glog"
+)
+
+// Reporter describes a mechanism for recording compliance incidents.
+type Reporter interface {
+	// Log records an incident with the given parameters.
+	// The baseURL, summary and category fields should be stable across
+	// multiple similar incidents to allow aggregation.  Information that
+	// varies between instances of the 'same' incident should be included in
+	// the fullURL or details field.
+	Log(ctx context.Context, baseURL, summary, category, fullURL, details string)
```

Comment:
Will all incidents have a fullURL? MMD probes, for instance, use a number of API endpoints before they may ultimately record an incident. What would the fullURL be in that case?

Reply:
Yeah, I wouldn't expect every incident to have the full URL, but I think there are enough cases where it's really useful (`https://broken.log/ct/v1/get-sth-consistency?from=10&to=20`) that it's worth having a specific field for it.